lib/path.js

```diff
@@ -140,6 +140,24 @@ function normalizeStringPosix(path, allowAboveRoot) {
   return res;
 }
 
+function _format(sep, pathObject) {
+  if (pathObject === null || typeof pathObject !== 'object') {
+    throw new TypeError(
```

Comment:
Oh, good points about the stack trace and polymorphism as well. Rebased against master, changed according to these comments, and force pushed.